processor/src/main/java/org/mapstruct/ap/internal/util/Executables.java

```diff
@@ -190,19 +251,29 @@ private static void addEnclosedElementsInHierarchy(Elements elementUtils, List<E
      * @param methodsToAdd methods to add to alreadyAdded, if they are not yet overridden by an element in the list
      * @param parentType the type for with elements are collected
      */
-    private static void addNotYetOverridden(Elements elementUtils, List<ExecutableElement> alreadyCollected,
+    private static void addNotYetOverridden(Elements elementUtils, List<Accessor> alreadyCollected,
                                             List<ExecutableElement> methodsToAdd, TypeElement parentType) {
-        List<ExecutableElement> safeToAdd = new ArrayList<ExecutableElement>( methodsToAdd.size() );
+        List<Accessor> safeToAdd = new ArrayList<Accessor>( methodsToAdd.size() );
         for ( ExecutableElement toAdd : methodsToAdd ) {
             if ( isNotObjectEquals( toAdd )
                 && wasNotYetOverridden( elementUtils, alreadyCollected, toAdd, parentType ) ) {
-                safeToAdd.add( toAdd );
+                safeToAdd.add( new ExecutableElementAccessor( toAdd ) );
             }
         }
 
         alreadyCollected.addAll( 0, safeToAdd );
     }
 
+    private static void addFields(List<Accessor> alreadyCollected, List<VariableElement> variablesToAdd) {
+        List<Accessor> safeToAdd = new ArrayList<Accessor>( variablesToAdd.size() );
+        for ( VariableElement toAdd : variablesToAdd ) {
+            //TODO should we do some check?
```

Comment:
I was thinking something like with the overriden methods. Whether we should check if a field is already added (shadowed). However, I think it is not needed, as later on the first found field will be used and the rest will be ignored